plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1953,13 +2048,17 @@ private Optional<Long> getSnapshotId(Table table, Optional<Long> snapshotId, boo
     {
         // table.name() is an encoded version of SchemaTableName
         return snapshotId
-                .map(id ->
-                        snapshotIds.computeIfAbsent(
-                                table.name() + "@" + id,
-                                ignored -> IcebergUtil.resolveSnapshotId(table, id, allowLegacySnapshotSyntax)))
+                .map(id -> resolveSnapshotId(table, id, allowLegacySnapshotSyntax))
                 .or(() -> Optional.ofNullable(table.currentSnapshot()).map(Snapshot::snapshotId));
     }
 
+    private long resolveSnapshotId(Table table, long id, boolean allowLegacySnapshotSyntax)
+    {
+        return snapshotIds.computeIfAbsent(
+                table.name() + "@" + id,
+                ignored -> IcebergUtil.resolveSnapshotId(table, id, allowLegacySnapshotSyntax));
```

Comment:
`allowLegacySnapshotSyntax` should be part of the cache  key

(pre-existing, but still sth to fix)

Reply:
https://github.com/trinodb/trino/blob/9b7c1b95bf1d76ee4a6c87bfcb502667c8b7f6ca/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java#L221-L232

In case that `allowLegacySnapshotSyntax` is `false` we'd get a Trino exception.

I don't see the purpose behind this request.

Reply:
> In case that `allowLegacySnapshotSyntax` is `false` we'd get a Trino exception.

only if `snapshotIds` entry doesn't exist yet.
cc @phd3